iOSMoviesApp/Controllers/DiscoveryPage/DiscoveryPageViewController.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  DiscoveryPageViewController.swift
+//  iOSMoviesApp
+//
+//  Created by Tarang Sultania on 08/07/25.
+//
+
+import Foundation
+import UIKit
+import CoreLocation
+
+protocol DiscoveryPageViewControllerDelegate: AnyObject{
+    func reloadTableView()
+    func navigateToDetails(for indexPath : IndexPath)
+}
+
+class DiscoveryPageViewController: UIViewController{
+    
+    @IBOutlet weak var locationBtn: UIButton!
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    private var movieListVM : MovieListViewModel!
+    let locationService : LocationService = LocationService()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupLocationService()
+        setupUI()
+    }
+    
+    private func setupUI() {
+        movieListVM = MovieListViewModel(delegate : self)
+        movieListVM.fetchMovies()
+        setupTableView()
+        setupSearchBar()
+    }
+    
+    func setupTableView() {
+        self.tableView.register(UINib(nibName: "MovieCardTableViewCell", bundle: nil), forCellReuseIdentifier: "MovieCardTableViewCell")
+        self.tableView.delegate = self
```

Comment:
```suggestion
        self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "DefaultCell")
        self.tableView.delegate = self
```
Group similar code together in one place to improve readability and maintainability.